src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,252 @@
+/**
+ * @file cmaes.hpp
+ * @author Marcus Edel
+ * @author Kartik Nighania
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+namespace mlpack {
+namespace optimization {
+
+/**
+*CMA-ES stands for Covariance Matrix Adaptation Evolution Strategy.
+*Evolution strategies (ES) are stochastic, derivative-free methods for 
+*numerical optimization of non-linear or non-convex continuous optimization
+*problems. They belong to the class of evolutionary algorithms and 
+*evolutionary computation.
+*
+*An evolutionary algorithm is broadly based on the principle
+*of biological evolution, namely the repeated interplay
+*of variation (via recombination and mutation) and selection: in each 
+*generation (iteration) new individuals (candidate solutions, denoted as
+*x) are generated by variation, usually in a stochastic
+*way, of the current parental individuals. Then, some individuals are
+*selected to become the parents in the next generation based on their
+*fitness or objective function value f(x). Like
+*this, over the generation sequence, individuals with better and better
+*f-values are generated.
+*/
+class CMAES
+{
+ public:
+/**
+* Constructor for the CMAES optimizer.
+* 
+* @param objectDim the dimension of the object
+* @param start the initial start point of the optimizer
+*        double value
+* @param stdDivs the intial standard deviation to choose for the
+*        gaussian distribution.
+* @param iters the maximum number of iterations to reach the minimum.
+*        it may happen that the function gets terminated by reaching 
+*        the condition and not using the remaining iterations
+* @param evalDiff the change in function value to see if flat fitness
+*        is matched which is the condition mostly when minima is reached.
+* @param functionHistory check for minimum function value difference from
+         the history of function values to check for flat fitness. 
+*/ 
+CMAES(int objectDim = 0,
```

Comment:
Missing space here, also can you align the parameter with the first:

```c++
 CMAES(const size_t objectDim = 0,
       const double start = 0,
              ...);
```


Also if you could use ``const`` and ``size_t`` instead of ``int`` to be consistent with the rest of the codebase here, that would be much appreciated.